src/components/Vault/VaultDepositForm.tsx

```diff
@@ -91,17 +92,22 @@ const VaultDepositForm: FC<VaultProps> = (props) => {
   const waitApprove = useWaitForTransaction({
     hash: approve.data?.hash,
   })
-  // Preview deposit method
-  const { isLoading: isLoadingPreview } = useContractRead({
-    chainId,
-    abi: vaultCompounderAbi,
-    address: vaultAddress,
-    functionName: "previewDeposit",
-    args: [value],
-    onSuccess: (data) => {
-      form.setValue("amountOut", formatUnits(data, ybToken?.decimals || 18))
-    },
-  })
+
+  const { isLoading: isLoadingPreview, data: assetToYbToken } =
+    useAssetToYbToken({
+      chainId,
+      id: poolId,
+      token: inputTokenAddress,
+      amount: value.toString(),
+      type: props.type,
+    })
+
+  useEffect(() => {
```

Comment:
`useEffect` can be messy and should be avoided as much as possible. Ideally, you should use `onSuccess`; which you can pass to `useContractRead` via `useAssetToYbToken`.

Reply:
Added onSuccess and onError, onError is used to reset the preview field when the user erases all data in the input